src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -658,7 +662,7 @@ public static JsonObjectBuilder json(MetadataBlock metadataBlock, boolean printO
                 fieldsBuilder.add(datasetFieldType.getName(), json(datasetFieldType, ownerDataverse));
             }
         }
-
+        
         jsonObjectBuilder.add("fields", fieldsBuilder);
         return jsonObjectBuilder;
     }
```

Comment:
I'm just going to put this comment here at the bottom but @ffritze can you please get the following test passing?

edu.harvard.iq.dataverse.api.DataversesIT.testListMetadataBlocks is failing with this:

```
1 expectation failed.
JSON path data[1].fields.size() doesn't match.
Expected: <10>
  Actual: <2>
```

I also mentioned this here: https://dataverse.zulipchat.com/#narrow/stream/378866-troubleshooting/topic/metadatablocks.20api/near/460096243

Reply:
When I start mvn clean package from my pull request branch which has dataverse 6.3 as basis there are no failing tests:
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.020 s -- in edu.harvard.iq.dataverse.feedback.FeedbackUtilTest
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 1595, Failures: 0, Errors: 0, Skipped: 36
[INFO] 
[INFO] 
[INFO] --- jacoco:0.8.11:report (jacoco-after-unit) @ dataverse ---
[INFO] Loading execution data file /home/ff/github/dataverse/target/coverage-reports/jacoco-unit.exec
[INFO] Analyzed bundle 'dataverse' with 1044 classes
[INFO] 
[INFO] --- war:3.3.2:war (default-war) @ dataverse ---
[INFO] Packaging webapp
[INFO] Assembling webapp [dataverse] in [/home/ff/github/dataverse/target/dataverse]
[INFO] Processing war project
[INFO] Copying webapp resources [/home/ff/github/dataverse/src/main/webapp]
[INFO] Building war: /home/ff/github/dataverse/target/dataverse-6.3.war
[INFO] Packaging classes
[INFO] Building jar: /home/ff/github/dataverse/target/dataverse-6.3-classes.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:02 min
[INFO] Finished at: 2024-08-26T08:06:19+02:00
[INFO] ------------------------------------------------------------------------

That's why I am not quite sure if the failing test has something to do with my pull request. Should I switch to develop and try to fix the test anyway?

Reply:
@ffritze 

Those tests executed at build time are unit tests. @pdurbin is referring to the integration tests, and in particular the test `testListMetadataBlocks`, present in: `DataversesIT.java`.

To run integration tests, you need to have the environment running locally. You can do this by using the containerized dev environment.

Steps:

1) Use `mvn -Pct package` to build the application image.
2) Use `docker-compose -f docker-compose-dev.yml up` to run the containerized environment
3) When all containers are up and running, execute: `mvn test -Dtest=DataversesIT` and all the IT tests from DataversesIT will be executed.

I recommend that you update this branch with develop, as the metadata blocks endpoint was recently modified, and I'm not sure if you have the latest changes.

The test may fail because of the changes you have made. There are assertions in the test that check the number of fields returned, and your changes likely reduced this number by removing duplicate child fields that were previously listed (repeated) as parents.

Reply:
@GPortas thanks for the detailed instructions.

Tests are are passing now: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10764/9/testReport/

Thanks, @ffritze!